### DISCUSSION TITLE
Neovide prompt in insert mode doesnt change to a line

### DISCUSSION THREAD

**Discussion by kaushiknfs** [neovide/neovide#989] [Q&A]
Hi,
Neovide prompt in insert mode doesnt change to a line
![135137514-09adad4f-5a2c-4618-bf59-82b7e6d320b4](https://user-images.githubusercontent.com/369865/135399406-e7868147-181e-4d98-b78a-e688e2df8a56.png)

is this the default behavior? how do i change the insert mode cursor, standard vim way of set guicursor doesnt seem to work.
Please suggest.
Thanks,
.

**Comment by Kethku** [comment]
We parse the cursor mode here https://github.com/neovide/neovide/blob/main/src/bridge/events.rs#L438
Which is stored and passed to the renderer here https://github.com/neovide/neovide/blob/main/src/editor/mod.rs#L102-L107
Which in turn is passed to the cursor renderer here https://github.com/neovide/neovide/blob/main/src/renderer/mod.rs#L198

So my understanding is that this should be supported just fine. Plus on my machine without guicursor set, 
![insert cursor](https://user-images.githubusercontent.com/3323631/136294404-1a4d21d5-755d-4bfb-86f2-8050aca9343d.gif)

So maybe I'm missing something? Seems to work for me.

**Comment by kaushiknfs** [comment]
Thank you. not sure what am missing though.
